### PR TITLE
KOGITO-4094 Avoid to deploy artifacts if author is not `kiegroup` and no artifacts repository has been given

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-4094

Related PRs:
- https://github.com/kiegroup/kogito-runtimes/pull/962
- https://github.com/kiegroup/optaplanner/pull/1095
- https://github.com/kiegroup/kogito-apps/pull/590
- https://github.com/kiegroup/kogito-examples/pull/515

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: Jenkinsfile.deploy [hunk 1]
```diff
@@ -221,6 +229,10 @@ String getBotAuthorCredsID(){
     return params.BOT_CREDENTIALS_ID
 }
 
+boolean isKiegroupGitAuthor(){
```
Comment (rsynek) [inline Jenkinsfile.deploy:232]: Soft suggestion: maybe we should focus on the purpose; how about `isNotTestingBuild()` ?

**Review by rsynek — APPROVED** [review]

**Comment by radtriste** [comment]
tested all paths ok

**Comment by radtriste** [comment]
tested all paths ok

**Comment by radtriste** [comment]
jenkins rerun runtimes tests

**Review by rsynek — APPROVED** [review]

**Pull request merged**